src/hooks/useRoedmapCreate.ts

```diff
@@ -1,14 +1,42 @@
 import { useState } from 'react';
-import type { RoadmapInfo } from '@/pages/roadmap/create';
+import type { RoadmapInfo, Step } from '@/pages/roadmap/create';
 
-export const useRoeadmapCreate = (defaultValue: RoadmapInfo) => {
-  const [info, setInfo] = useState<RoadmapInfo>(defaultValue);
+export type RoadmapValid = 'roadmap' | 'step' | null;
 
-  const handleOnChange = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+export const useRoeadmapCreate = (defaultInfo: RoadmapInfo, defaultStep: Step) => {
+  const [info, setInfo] = useState<RoadmapInfo>(defaultInfo);
+  const [stepList, setStepList] = useState<Step[]>([]);
+  const [step, setStep] = useState<Step>(defaultStep);
+  const [valid, setValid] = useState<RoadmapValid>(null);
+
+  const handleInfo = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
     const { name, value } = e.target;
 
     setInfo((prev) => ({ ...prev, [name]: value === 'public' ? true : value === 'private' ? false : value }));
   };
```

Comment:
요 부분 
```
const handleInfo = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
    const { name, value } = e.target;

    if (name === 'isPublic'){
        setInfo({...prev, isPublic: value })
    }
    else{
        setInfo({...prev, [name]: value })
    }
}
```

위 처럼 바꾸는건 어떻게 생각하시나요? 지금 코드는 한줄이라 깔끔하긴하지만 이해하는데는 조금 걸리는 것 같아요.
```suggestion
const handleInfo = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
    const { name, value } = e.target;

    if (name === 'isPublic'){
        setInfo({...prev, isPublic: value })
    }
    else{
        setInfo({...prev, [name]: value })
    }
}
```

Reply:
반영했습니당~